packages/web3-eth-accounts/src/index.js

```diff
@@ -272,20 +297,120 @@ function _validateTransactionForSigning(tx) {
         );
     }
 
-    if (!tx.gas && !tx.gasLimit) {
+    if (
+        (!tx.gas && !tx.gasLimit) &&
+        (!tx.maxPriorityFeePerGas && !tx.maxFeePerGas)
+    ) {
         return new Error('"gas" is missing');
     }
 
-    if (tx.nonce < 0 ||
-        tx.gas < 0 ||
-        tx.gasPrice < 0 ||
-        tx.chainId < 0) {
-        return new Error('Gas, gasPrice, nonce or chainId is lower than 0');
+    if (tx.gas && tx.gasPrice) {
+        if (tx.gas < 0 || tx.gasPrice < 0) {
+            return new Error('Gas or gasPrice is lower than 0');
+        }
+    } else {
+        if (tx.maxPriorityFeePerGas < 0 || tx.maxFeePerGas < 0) {
+            return new Error('maxPriorityFeePerGas or maxFeePerGas is lower than 0');
+        }
+    }
+
+    if (tx.nonce < 0 || tx.chainId < 0) {
+        return new Error('Nonce or chainId is lower than 0');
     }
 
     return;
 }
 
+function _handleTxType(tx) {
+    // Taken from https://github.com/ethers-io/ethers.js/blob/2a7ce0e72a1e0c9469e10392b0329e75e341cf18/packages/abstract-signer/src.ts/index.ts#L215
+    const hasEip1559 = (tx.maxFeePerGas !== undefined || tx.maxPriorityFeePerGas !== undefined);
+
+    let txType;
+
+    if (tx.type !== undefined) {
+        txType = utils.toHex(tx.type)
+    } else if (tx.type === undefined && hasEip1559) {
+        txType = '0x2'
+    } else {
+        txType = '0x0'
+    }
+
+    if (tx.gasPrice !== undefined && (txType === '0x2' || hasEip1559))
+        throw Error("eip-1559 transactions don't support gasPrice");
+    if ((txType === '0x1' || txType === '0x0') && hasEip1559)
+        throw Error("pre-eip-1559 transaction don't support maxFeePerGas/maxPriorityFeePerGas");
+    
+    if (
+        hasEip1559 ||
+        (
+            (tx.common && tx.common.hardfork && tx.common.hardfork.toLowerCase() === HardForks.London) ||
+            (tx.hardfork && tx.hardfork.toLowerCase() === HardForks.London)
+        )
+    ) {
+        txType = '0x2';
+    } else if (
+        tx.accessList ||
+        (
+            (tx.common && tx.common.hardfork && tx.common.hardfork.toLowerCase() === HardForks.Berlin) ||
+            (tx.hardfork && tx.hardfork.toLowerCase() === HardForks.Berlin)
+        )
+    ) {
+        txType = '0x1';
+    }
+    
+    return txType
+}
+
+function _handleTxPricing(_this, tx) {
+    return new Promise((resolve, reject) => {
+        try {
+            if (tx.type < '0x2' && tx.gasPrice !== undefined) {
```

Comment:
when `tx.type = undefined` and is checked under `tx.type < "0x2"` it'll result into false
perhaps:
```
if ( tx.type === 'undefined' || tx.type < '0x2' && ...
 ```

Reply:
It's intended to be false if `tx.type` is `undefined`. We only want to resolve the provided `tx.gasPrice` if `tx.type` is `0x0` or `0x1`. If `undefined`, we don't know if the network supports `EIP-1559`, and whether we should use `tx.gasPrice` or the 1559 properties